cli/src/semgrep/core_runner.py

```diff
@@ -231,8 +231,19 @@ async def _core_stdout_processor(self, stream: asyncio.StreamReader) -> None:
                 # happens if the data that follows a sequence of zero
                 # or more ".\n" has fewer than two bytes, such as:
                 # "", "3", ".\n.\n3", ".\n.\n.\n.", etc.
+
+                # Hack: the exact wording of parts this message may be used in metrics queries
+                # that are looking for it. Make sure `semgrep-core exited with unexpected output`
+                # and `interfile analysis` are both in the message, or talk to Emma.
                 raise SemgrepError(
-                    "semgrep-core exited successfully but produced unexpected output"
+                    "semgrep-core exited with unexpected output.\n\n"
+                    "\tThis can happen because it overflowed the stack or used too much memory.\n"
+                    "\tTry changing the stack limit with `ulimit -s <limit>` or adding\n"
+                    "\t`--max-memory <memory>` to the semgrep command. For CI runs with interfile\n"
+                    "\tanalysis, the default max-memory is 5000MB. Depending on the memory\n"
+                    "\tavailable in your runner, you may need to lower it. We recommend choosing\n"
+                    "\ta limit 70% of the available memory to allow for some overhead. If you have\n"
+                    "\ttried these steps and still are seeing this error, please contact us."
```

Comment:
contact us at support@semgrep.dev? I think there is a constant in the codebase for it already